src/main/java/frc/robot/util/Node.java

```diff
@@ -0,0 +1,80 @@
+package frc.robot.util;
+
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Pose3d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj.DriverStation.Alliance;
+import frc.robot.constants.VisionConstants;
+
+/**Class to store data about scoring locations*/
+public class Node {
+    // Possible node types
+    public enum Types {CONE, CUBE, HYBRID};
+
+    public final Alliance alliance;
+    public final int row;
+    public final int column;
+    public final Pose3d pose;
+    public final Pose2d scorePose;
+    public final Types type;
+
+    /**
+     * Creates a new Node with default values
+     */
+    public Node(){
+        alliance = DriverStation.getAlliance();
+        row = 1;
+        column = 1;
+        pose = new Pose3d();
+        scorePose = new Pose2d();
+        type = Types.HYBRID;
+    }
+
+    /**
+     * Creates a new Node object
+     * @param team
+     *  The node's color
+     * @param row
+     *  Which row it's in (1 = bottom, 2 = middle, 3 = top)
+     * @param column
+     *  Grid column from left to right (1-9)
+     */
+    public Node(Vision vision, Alliance alliance, int row, int column){
+        this.alliance=alliance;
+        this.row=row;
+        this.column=column;
+        type = row==1?Types.HYBRID:column%3==2?Types.CUBE:Types.CONE;
+
+        // Which April tag this node is closest to
+        Pose3d tag = vision.getTagPose((column-1)/3+(alliance==Alliance.Red?1:6));
```

Comment:
I don't think holding the node positions relitive to the tag is the best way to do it. Additional  maybe this file and the align command should be in a seprate PR